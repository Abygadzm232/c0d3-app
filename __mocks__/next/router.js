global.window = Object.create(window)
Object.defineProperty(global.window, 'location', {
  value: { pathname: '/not-root' } // make sure pathname isnt '/' by default
})
const push = jest.fn(path => (global.window.location['pathname'] = path))
export const useRouter = jest.fn().mockImplementation(() => {
  return {
    route: '/',
    pathname: '/',
    query: {},
    asPath: '/',
    push: push,
    replace: async () => true,
    reload: () => null,
    back: () => null
  }
})
