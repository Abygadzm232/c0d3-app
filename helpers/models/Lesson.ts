import { DataTypes, Model } from 'sequelize'

export class Lesson extends Model {
  public id!: number
  public description!: string
  public docUrl?: string
  public githubUrl?: string
  public videoUrl?: string
  public order!: number
  public titel!: string
  public chatUrl?: string

  public readonly createdAt!: Date
  public readonly updatedAt!: Date
}

export const LessonTypes = {
  description: DataTypes.TEXT,
  docUrl: DataTypes.STRING,
  githubUrl: DataTypes.STRING,
  videoUrl: DataTypes.STRING,
  order: DataTypes.INTEGER,
  title: DataTypes.STRING,
  chatUrl: DataTypes.STRING
}
