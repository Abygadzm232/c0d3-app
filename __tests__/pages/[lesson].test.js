jest.mock('next/router')
jest.mock('@apollo/react-hooks')
jest.mock('../../components/LessonCard')
import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { useRouter } from 'next/router'
import Lesson from '../../pages/curriculum/[lesson]'
import { render } from '@testing-library/react'

describe('Lesson Page', () => {
  useQuery.mockReturnValue({
    data: {
      lessons: [
        {
          id: '5',
          title: 'Foundations of JavaScript',
          description: 'A super simple introduction to help you get started!',
          docUrl:
            'https://www.notion.so/garagescript/JS-0-Foundations-a43ca620e54945b2b620bcda5f3cf672',
          githubUrl: '',
          videoUrl:
            'https://www.youtube.com/watch?v=H-eqRQo8KoI&list=PLKmS5c0UNZmewGBWlz0l9GZwh3bV8Rlc7&index=1',
          order: 0,
          challenges: [
            {
              id: '107',
              title: 'Sum of 2 Numbers',
              description:
                "Write a function that takes in 2 numbers and returns their sum.\n\nHere's how another developer might use your function:\n\n```\nsolution(5,9) // Should return 14\nsolution(4,1) // Should return 5\n```",
              order: 1,
              __typename: 'Challenge'
            },
            {
              id: '108',
              title: 'Sum of 3 Numbers',
              description:
                "Write a function that takes in 3 numbers and returns their sum.\n\nHere's how another developer might use your function:\n\n```\nsolution(5,9,2) // Should return 16\nsolution(4,1,9) // Should return 14\n```",
              order: 2,
              __typename: 'Challenge'
            }
          ]
        }
      ],
      session: {
        user: {
          username: 'test'
        },
        lessonStatus: [
          {
            lessonId: '4',
            isPassed: '1235435',
            isEnrolled: '123456'
          }
        ],
        submissions: [
          {
            id: '6050',
            status: 'open',
            mrUrl: null,
            diff:
              '\u001b[1mdiff --git a/index.js b/index.js\u001b[m\n\u001b[1mindex 5e1c309..0929f9d 100644\u001b[m\n\u001b[1m--- a/index.js\u001b[m\n\u001b[1m+++ b/index.js\u001b[m\n\u001b[36m@@ -1 +1 @@\u001b[m\n\u001b[31m-Hello World\u001b[m\n\\ No newline at end of file\u001b[m\n\u001b[32m+\u001b[m\u001b[32mHello to the new c0d3 CLI\u001b[m\n',
            viewCount: 0,
            comment: null,
            order: null,
            challengeId: '108',
            lessonId: '5',
            reviewer: null,
            createdAt: '1589315948051',
            updatedAt: '1589315948098',
            __typename: 'Submission'
          },
          {
            id: '6048',
            status: 'open',
            mrUrl: null,
            diff:
              '\u001b[1mdiff --git a/index.js b/index.js\u001b[m\n\u001b[1mindex 5e1c309..0929f9d 100644\u001b[m\n\u001b[1m--- a/index.js\u001b[m\n\u001b[1m+++ b/index.js\u001b[m\n\u001b[36m@@ -1 +1 @@\u001b[m\n\u001b[31m-Hello World\u001b[m\n\\ No newline at end of file\u001b[m\n\u001b[32m+\u001b[m\u001b[32mHello to the new c0d3 CLI\u001b[m\n',
            viewCount: 0,
            comment: null,
            order: null,
            challengeId: '143',
            lessonId: '2',
            reviewer: null,
            createdAt: '1589247516191',
            updatedAt: '1589316605068',
            __typename: 'Submission'
          }
        ]
      },
      alerts: [
        {
          id: '0',
          text: 'Set up your computer to submit challenges.',
          type: 'info',
          url:
            'https://www.notion.so/JS-0-Foundations-a43ca620e54945b2b620bcda5f3cf672#b45ed85a95e24c9d9fb784afb7a46bcc',
          urlCaption: 'View Instructions'
        },
        {
          id: '1',
          text: 'Please upgrade your CLI client by running npm update c0d3.',
          type: 'urgent'
        }
      ]
    }
  })
  test('Should render lesson page for students', async () => {
    useRouter.mockReturnValue({
      query: {
        lesson: '5'
      }
    })
    const { container } = render(<Lesson />)
    expect(container).toMatchSnapshot()
  })
})
