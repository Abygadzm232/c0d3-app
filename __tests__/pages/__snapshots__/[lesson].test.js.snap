// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson Page Should render lesson page for students 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand text-primary font-weight-bold"
          href="/"
        >
          C0D3
        </a>
        <div
          id="navbarNav"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <div
              class="navbar-nav collapse navbar-collapse"
            >
              <a
                class="nav-item nav-link active"
                href="/"
              >
                <span
                  class="sr-only"
                >
                  (current)
                </span>
                Home
              </a>
              <a
                class="nav-item nav-link"
                href="/#learning"
              >
                Learning Process
              </a>
              <a
                class="nav-item nav-link"
                href="https://c0d3.com/book"
                rel="noopener noreferrer"
                target="_blank"
              >
                Resources
              </a>
              <a
                class="nav-item nav-link"
                href="https://chat.c0d3.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Help
              </a>
              <a
                class="nav-item nav-link"
                href="/contributors"
              >
                Contributors
              </a>
            </div>
          </div>
        </div>
        <div>
          <a
            class="btn btn-secondary border mr-3"
            href="/login"
          >
            Login
          </a>
          <a
            class="btn btn-secondary border mr-3"
            href="/signup"
          >
            Signup
          </a>
        </div>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="row mt-4"
      >
        <div
          class="challenges-container"
        >
          <div
            class="card shadow-sm mt-3 col-12 px-0 pt-3 border-0"
          >
            <div
              class="card-body p-0"
            >
              <div
                class="d-flex mb-3 px-3"
              >
                <img
                  alt="lesson-cover"
                  class="lessonTitleCard__lesson-cover mr-3"
                  src="/curriculumAssets/lessonCoversSvg/js-0-cover.svg"
                />
                <div>
                  <p
                    class="m-0"
                  >
                    <a
                      class=""
                      href="/curriculum"
                    >
                      Go Back
                    </a>
                  </p>
                  <h1
                    class="lessonTitleCard__lesson-title"
                  >
                    Foundations of JavaScript
                  </h1>
                </div>
              </div>
              <div
                class="card-footer bg-white p-0"
              >
                <a
                  class="btn border-right rounded-0 px-4 py-3"
                  href="https://www.notion.so/garagescript/JS-0-Foundations-a43ca620e54945b2b620bcda5f3cf672"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  LESSON
                </a>
                <a
                  class="btn border-right rounded-0 px-4 py-3"
                  href="#"
                >
                  CHALLENGES
                </a>
              </div>
            </div>
          </div>
          <div
            class="alerts-container "
          >
            <div
              class="alert d-flex justify-content-between mt-3 bg-primary text-white"
              role="alert"
            >
              <div>
                <img
                  class="mr-3 alert-icon"
                  src="/curriculumAssets/icons/icon-tip.svg"
                />
                Set up your computer to submit challenges. 
                <a
                  class="text-white"
                  href="https://www.notion.so/JS-0-Foundations-a43ca620e54945b2b620bcda5f3cf672#b45ed85a95e24c9d9fb784afb7a46bcc"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Instructions
                </a>
              </div>
              <button
                class="alert-dismiss"
                data-testid="dismiss-info"
                role="dismiss"
              >
                <img
                  src="/curriculumAssets/icons/dismiss-info.svg"
                />
              </button>
            </div>
            <div
              class="alert d-flex justify-content-between mt-3 alert-danger"
              role="alert"
            >
              <div>
                <img
                  class="mr-3 alert-icon"
                  src="/curriculumAssets/icons/exclamation.svg"
                />
                Please upgrade your CLI client by running npm update c0d3. 
              </div>
              <button
                class="alert-dismiss"
                data-testid="dismiss-urgent"
                role="dismiss"
              >
                <img
                  src="/curriculumAssets/icons/dismiss-urgent.svg"
                />
              </button>
            </div>
          </div>
          <div
            class="row challenge-display mt-3"
          >
            <div
              class="col-4"
            >
              <div
                class="card mb-2 challenge-title-card challenge-title-card--active shadow-sm border-0"
                data-testid="challenge-title"
              >
                <div
                  class="card-body d-flex justify-content-between"
                >
                  <div>
                    1. Sum of 2 Numbers
                  </div>
                </div>
              </div>
              <div
                class="card mb-2 challenge-title-card shadow-sm border-0"
                data-testid="challenge-title"
              >
                <div
                  class="card-body d-flex justify-content-between"
                >
                  <div>
                    2. Sum of 3 Numbers
                  </div>
                  <img
                    height="25px"
                    src="/curriculumAssets/icons/pending.svg"
                    width="25px"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-8"
            >
              <div
                class="card shadow-sm border-0"
              >
                <div
                  class="card-body challenge-question"
                >
                  <h1
                    class="card-title"
                    data-testid="challenge-question-title"
                  >
                    Sum of 2 Numbers
                  </h1>
                  <div
                    class="bg-light p-3 mt-3"
                    data-testid="challenge-question-description"
                  >
                    <div>
                      <p>
                        Write a function that takes in 2 numbers and returns their sum.
                      </p>
                      <p>
                        Here's how another developer might use your function:
                      </p>
                      <pre>
                        <code>
                          solution(5,9) // Should return 14
solution(4,1) // Should return 5
                        </code>
                      </pre>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="mt-5 font-weight-light text-center pb-5 text-muted"
    >
      Â© Copyright 2020 GarageScript
      <br />
      All Rights Reserved
    </footer>
  </div>
</div>
`;
